### PR TITLE
Requests & Responses examples for Public API

### LINKED ISSUE #3191: Add HTTP examples for Public API to documentation
* Add CURL / HTTP examples for all documented API's
* Include example JSON request / responses

**Relates to:** 

* https://github.com/gitpod-io/website/issues/3174

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
## Description

cURL Requests & Responses examples for Public API

## Related Issue(s)

Fixes #3191

## How to test

- Go To [**`/docs/references/gitpod-public-api`**](https://deploy-preview-3196--gitpod-kumquat.netlify.app/docs/references/gitpod-public-api) & see :eyes:

## Release Notes

```release-note
Docs: Public API _(Alpha)_
```

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3196"></a>

### CODE REVIEW

File: gitpod/docs/references/gitpod-public-api.md [hunk 1]
```diff
@@ -15,7 +19,20 @@ The Gitpod Public API grants direct access to Gitpod data for use cases such as
 
 ## Authentication
 
-To access the public API, you will need to pass an [Access Token](/docs/configure/user-settings/access-tokens) as a an Authorization request header bearer token in the format `Bearer <your-access-token>`.
+To authenticate requests, the Gitpod API utilises [Access Token](/docs/configure/user-settings/access-tokens). In the [Gitpod Settings](https://gitpod.io/tokens), you can view and manage your PAT Tokens (_Private Beta_).
+
+Keep in mind that your access tokens have a lot of power, so protect them carefully! Avoid disclosing your private Access Tokens in areas that are open to the public, such as GitHub, client-side code, and so forth.
+
+To Authenticate you will need to pass an Access Token as an Authorization request header bearer token in the format `Bearer <your-access-token>`.
```
Comment (Siddhant-K-code) [inline gitpod/docs/references/gitpod-public-api.md:26]:
```suggestion
To Authenticate you will need to pass an Access Token as an Authorization request header bearer token in the format `Bearer <YOUR_PAT_TOKEN>`.
```

File: gitpod/docs/references/gitpod-public-api.md [hunk 2]
```diff
@@ -263,7 +730,34 @@ Joins a team.
 
 Resets the team invitation.
 
-**Usage**: `resetTeamInvitation()`
+<APIToggle id="api-toggle">
```
Comment (loujaybee) [inline gitpod/docs/references/gitpod-public-api.md:733]: I'd suggest to expand and leave these open, I think seeing the CURL example is what users would want, so better to show that information by default.

File: src/lib/components/docs/api-toggle.svelte [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    {
+      mobileTitle: "Go",
+      title: "Go ",
+      value: 2,
+      slotName: "goExample",
+      hidden: false,
+    },
+    {
+      mobileTitle: "Example Response",
+      title: "Example Response",
```
Comment (loujaybee) [inline src/lib/components/docs/api-toggle.svelte:22]: Could we swap to simply response? I suppose example is implied by the data.

```suggestion
      mobileTitle: "Response",
      title: "Response",
```

Comment (loujaybee) [reply src/lib/components/docs/api-toggle.svelte:22]: I'm also torn that this should be in a tab. I almost feel like I'd want to see this information by default, also.

File: gitpod/docs/references/gitpod-public-api.md [hunk 4]
```diff
@@ -15,7 +19,20 @@ The Gitpod Public API grants direct access to Gitpod data for use cases such as
...
+Example cURL command:
+
+```bash title="cURL"
+curl 'https://api.gitpod.io/gitpod.experimental.v1.TeamsService/ListTeams' \
+  -H 'content-type: application/json' \
+  -H 'Authorization: Bearer <YOUR_ACCESS_TOKEN>' \
+  --data '{}'
+```
 
 ## API language clients
```
Comment (loujaybee) [inline gitpod/docs/references/gitpod-public-api.md:37]: Can we move this section to the bottom? Keep curl up top, clients at the bottom?

File: gitpod/docs/references/gitpod-public-api.md [hunk 5]
```diff
@@ -177,7 +358,44 @@ Stops a running workspace (instance).
...
+```bash title="cURL"
+curl 'https://api.gitpod.io/gitpod.experimental.v1.WorkspacesService/DeleteWorkspace' \
+  -H 'content-type: application/json' \
+  -H 'Authorization: Bearer <YOUR_ACCESS_TOKEN>' \
+  --data '{"workspaceId":"<WORKSPACE_ID>"}'
+```
+
+</div>
+
+<div slot="responseExample">
```
Comment (loujaybee) [inline gitpod/docs/references/gitpod-public-api.md:380]: Can we comment out these empty responses for now?

File: gitpod/docs/references/gitpod-public-api.md [hunk 6]
```diff
@@ -102,7 +185,80 @@ Lists all the workspaces belongs to the authenticated user.
...
+    "description": "gitpod-io/empty ",
+    "status": {
+      "instance": {
+        "instanceId": "<INSTANCE_ID>",
+        "workspaceId": "<WORKSPACE_ID>",
+        "createdAt": "2022-12-22T07:43:16.152Z",
+        "status": {
+          "statusVersion": "<STATUS_VERSION>",
+          "phase": "PHASE_RUNNING",
+          "conditions": { "firstUserActivity": "2022-12-22T07:43:21.732Z" },
```
Comment (loujaybee) [inline gitpod/docs/references/gitpod-public-api.md:247]: **nit:** Bump response onto new line.

**Pull request merged**